doc/api/connection.md

```diff
@@ -0,0 +1,158 @@
+  of connections (optional, but either server or client is required).
+
+Don't call this constructor manually unless you use custom tranport.
+Recommended approach is to call `connect()` function provided by these modules:
+* `net`
+* `tls`
+* `ws`
+* `wss`
+
+### Methods
+
+#### connection.handshake(app, \[login\], \[password\], callback)
+
+* app: `String || Object` — application to connect to as `'name'` or
+  `'name@version'` or `{ name, version }`, where version must be
+  a valid semver range.
+* login: `String` — user name.
+* password: `String` — user password.
+* callback(error, sessionId) — callback function to invoke after the handshake
+  is completed.
+  * error: `Error`.
+  * sessionId: `number`.
+
+Send a handshake message over the connection.
+
+#### connection.inspectInterface(interfaceName, callback)
+
+* interfaceName: `String` — name of an interface to inspect.
+* callback(error, proxy) — callback function to invoke after another side responds
+  interface introspection.
+  * error: `Error`.
+  * proxy: `RemoteProxy` — remote proxy to the interface.
+
+Send an inspect message over the connection.
+
+#### connection.callMethod(interfaceName, methodName, args, callback)
+
+* interfaceName: `String` — name of an interface.
+* methodName: `String` — name of a method.
+* args: `Array` — method arguments.
+* callback(error, ...args) — callback function that is invoked after a callback
+  message has been received.
+  * error: `Error`.
+
+Send a call message over the connection.
+
+#### connection.emitRemoteEvent(interfaceName, eventName, args)
+
+* interfaceName: `String` — name of an interface.
+* eventName: `String` — name of an event.
+* args: `Array` — event arguments as an array.
+
+Send an event message over the connection.
+
+#### connection.ping(callback)
+
+* callback(error, ...args) — callback function to invoke after another side responds
+  with a pong message.
+  * error: `Error`.
+
+Send a ping message.
+
+#### connection.pong()
+
+Send a pong message.
+
+#### connection.startHeartbeat(interval)
+
+* interval: `number` — heartbeat interval in milliseconds.
+
+Start sending heartbeat messages.
+
+#### connection.stopHeartbeat()
+
+Stop sending heaarbeat messages.
+
+#### connection.close()
+
+Close the connection.
+
+#### connection.getTransport() `Object`
+
+Returns underlying transport.
+
+### Properties
+
+#### connection.server `Server`
+#### connection.client `Client`
+#### connection.id `number`
+#### connection.remoteAddress `Object`
+#### connection.handshakeDone `boolean`
+#### connection.username `string`
+#### connection.sessionId `number`
+#### connection.application `Application`
+#### connection.remoteProxies `Map of RemoteProxy`
+
+## Transport
+
+An abstract socket interface that specifies methods needed to send jstp messages.
+
+### Methods
+
+### transport.on(event, handler)
+
+* event: `string` — name of an event.
+* handler(...args).
+
+Used to subscribe on this events:
```

Comment:
I think we should use the same style as in `Connection` here and in other `Transport` methods as this is just another class description. And in this place specifically, I think it's better to just state that `Transport` must emit the following events and avoid describing `on` method (Maybe just state that it has to be an `EventEmitter`).